sn_networking/src/replication_fetcher.rs

```diff
@@ -56,12 +56,37 @@ impl ReplicationFetcher {
     ) -> Vec<(PeerId, RecordKey)> {
         self.remove_stored_keys(locally_stored_keys);
 
+        let new_data_key_opt = if incoming_keys.len() == 1 {
+            Some((
+                incoming_keys[0].0.to_record_key(),
```

Comment:
can we break this apart into some named variables, I'm not sure what's going on here.